modules/pam_listfile/pam_listfile.c

```diff
@@ -53,17 +53,16 @@ pam_sm_authenticate (pam_handle_t *pamh, int flags UNUSED,
     const char *citemp;
     char *ifname=NULL;
     char aline[256];
-    char mybuf[256],myval[256];
+    char mybuf[256],myval[256],apply_val[256];
     struct stat fileinfo;
     FILE *inf;
-    const char *apply_val;
     int apply_type;
 
     /* Stuff for "extended" items */
     struct passwd *userinfo;
 
     apply_type=APPLY_TYPE_NULL;
-    apply_val="";
+    memset(apply_val,'\0',sizeof(apply_val));
```

Comment:
Looks like zeroing out the whole `apply_val` buffer is superfluous, `apply_val[0] = '\0'` should be sufficient.